app/models/setting/puppet.rb

```diff
@@ -12,7 +20,7 @@ def self.default_settings
       self.set('enc_environment', N_("Foreman will explicitly set the puppet environment in the ENC yaml output. This will avoid conflicts between the environment in puppet.conf and the environment set in Foreman"), true, N_('ENC environment')),
       self.set('use_uuid_for_certificates', N_("Foreman will use random UUIDs for certificate signing instead of hostnames"), false, N_('Use UUID for certificates')),
       self.set('update_environment_from_facts', N_("Foreman will update a host's environment from its facts"), false, N_('Update environment from facts')),
-      self.set('update_subnets_from_facts', N_("Foreman will update a host's subnet from its facts"), false, N_('Update subnets from facts')),
+      self.set('update_subnets_from_facts', N_("Foreman will update a host's subnet from its facts"), 'all', N_('Update subnets from facts'), nil, { :collection => Proc.new { self.update_subnets_from_facts_types } }),
```

Comment:
The default value here should be "none". See my other remarks about inconsistency.